File: app/src/main/java/com/example/mytask/TaskDao.java
package com.example.mytask;


import java.util.List;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface TaskDao {

    //Firstdtable
    @Query("SELECT * FROM Productfile")
    List<Productfile> getAllProducts();

    @Insert
    void insert(Productfile task);


    @Query("UPDATE Productfile SET count = :Count,quantity = :quantity WHERE name =:Name AND Pid=:id")
    void updateProduct(Integer Count, String Name, String id,int quantity);





}

File: app/src/main/java/com/example/mytask/RoomDB/TaskDao.kt
package com.example.mytask.RoomDB;


import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.example.mytask.model.Productfile

@Dao
interface TaskDao {
    @Query("SELECT * FROM Productfile")
    fun getAllProducts(): List<Productfile>

    @Insert
    fun insert(task: Productfile?)


    @Query("UPDATE Productfile SET count = :Count,quantity = :quantity WHERE name =:Name AND regid=:id")
    fun updateProduct(Count: Int?, Name: String?, id: Int, quantity: Int)


    @Query("SELECT * FROM Productfile WHERE regid=:regId")
    fun getParticular(regId: Int): List<Productfile>?
}